dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -276,7 +276,7 @@ private ListenableFuture<Response> handleThrowable(Throwable clientSideThrowable
                 if (requestCanBeRetried() && shouldAttemptToRetry(clientSideThrowable)) {
                     callsiteStacktrace.ifPresent(clientSideThrowable::addSuppressed);
                     Meter retryReason = retryDueToThrowable.apply(clientSideThrowable);
-                    long backoffNanoseconds = getBackoffNanoseconds();
+                    long backoffNanoseconds = failures <= 1 ? 0 : getBackoffNanoseconds();
```

Comment:
We should have a test for this, similar to the [existing test](https://github.com/palantir/dialogue/blob/1f21912856b571a71a3a896e61f2a65efec760d5/dialogue-core/src/test/java/com/palantir/dialogue/core/RetryingChannelTest.java#L347-L349) for the [similar logic that handles 308 responses](https://github.com/palantir/dialogue/blob/1f21912856b571a71a3a896e61f2a65efec760d5/dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java#L315).

Reply:
Added.